### PR TITLE
Add validation by URL checking

### PR DESCRIPTION
**Pull request by ascheman**
Hi @marchof,

I came across some handles of Fediverse which turned out to be wrong (cf. 106d673).
Maybe this is due to the fact that people are new on Mastodon and are moving around to other servers?
Hopefully this will become more stable in the future.

Nevertheless, I introduced some advanced validation.
I didn't find a nice solution for Twitter so far (wanted to avoid calling the API for now).

Validation might take some time if the list becomes longer.
In particular there are some Fediverse servers which are really slow and need some retries every now and then (mostly the Thoughtworks server).
I have already some idea to handle this by 
* some validation caching which could also be handled on Github (by persisting the cache), 
* parallelise http requests 
but first wanted to check with you if you like the idea in general?

BTW: It would be great if you could at least introduce an `.editorconfig` to make refactoring and code formatting more concise.

Regs
  Gerd

### CODE REVIEW

**Comment by ascheman** [comment]
Looks like the check has failed. Besides the Thoughtworks server I experienced some instability of `mstdn.social` (which is currently only used by @BraunSusanne on this list). Will try to add some more robustness wrt. connection errors.

**Comment by marchof** [comment]
@ascheman I already had the same idea. But I would create a separate Main for it and have a separate scheduled workflow to not block manual updates.

**Comment by marchof** [comment]
@ascheman Some more ideas:

* All handles should be validated even if more than one fails
* A machine readable list of failures would be nice (e.g. json)
* The result can be integrated on the html page by marking invalid handles (some JavaScipts which loads the file from the previous step)

**Comment by marchof** [comment]
@ascheman One more thing:

* to play nice and not hit any limits the requests per host per second should be limited (e.g. max 1 request per host per second)

File: generator/src/main/java/org/javabubble/generator/model/ModelLoader.java [hunk 1]
```diff
@@ -13,23 +13,23 @@
...
 public class ModelLoader {
 
-	private final Path base;
+	private final Path path;
 	private final ObjectMapper mapper;
 
-	public ModelLoader(Path base) {
-		this.base = base;
+	public ModelLoader(Path path) {
+		this.path = path;
```
Comment (marchof) [inline generator/src/main/java/org/javabubble/generator/model/ModelLoader.java:20]: The idea here was that the model may be loaded from multiple yaml files in the future.

**Comment by ascheman** [comment]
Thanks for your feedback, @marchof 
Meanwhile I have done some more refactoring:
* The default behaviour of Main is again to validate only against the regex patterns,
* If you want to run validation against Github and Mastodon, run the main class with `--validate-on-platforms` CLI parameter.

However, the new code is still somewhat ugly (and still shows some Sonar issues), e.g., forwarding the validation-on-platforms parameter as argument through several methods. I would like to make the ModelValidator a class with less static methods and provide this as instance variable instead.

Maybe you could merge my changes, and we could treat more ideas and requirements as separate issues.

File: generator/src/main/java/org/javabubble/generator/model/ModelLoader.java (hunk 1 above)
Comment (ascheman) [reply generator/src/main/java/org/javabubble/generator/model/ModelLoader.java:20]: I see, but this allows better testability with small test sets for the moment. 
Should we refactor to a base directory with separate files if this idea becomes more clear?

**Comment by marchof** [comment]
I would prefer to have a separate package and Main class for the handle lookup. Then we keep aspects separately and can improve them independently.

**Comment by ascheman** [comment]
> I would prefer to have a separate package and Main class for the handle lookup. Then we keep aspects separately and can improve them independently.

Would you mind to outline the structure (e.g., by refactoring my code on a separate branch)? 
Currently I see some common properties for both packages/handlers, i.e., the regex patterns to validate (locally) and to derive target URLs for platform validation? How would you like to keep them in common? Derive from a common base class (in a common base package)? Provide them as static constants? To me this seems to be a key question wrt. further refactoring and split up into several packages.

**Comment by ascheman** [comment]
BTW: I have even more ideas as well, that was the reason to dive so deeply into your work. I would like to filter the list against the list of Twitter handles a persons currently follows. This would require Twitter API access. In the simplest case one could execute the code locally and provide the token by CLI (I mean, we are all Java programmers and should be able to do this).

**Comment by marchof** [comment]
Another feature I'm thinking about is pulling profiles infos from the social platforms. Probably we then get invalid handes as a side effect.

**Comment by marchof** [comment]
Hi @ascheman sorry for not following up on this. Just to busy and soon on vacation. What I learned over the last days is that for ther regular push action we need something very quick, as I check around 5 PRs a day. So in any case these checks needs to move to a asynchronous (e.g. nightly) action.

**Comment by marchof** [comment]
Hi @ascheman, I finally found some time to work on a solution (3eb3b319b0323e44cd7d754aa8395f9f904b35d2) which should fit in my workflow:

* It uses the Rest APIs of the respective platforms. This also allows to fix handles (casing)
* It is a scheduled action which runs every night independently from the generator. This keeps the generator fast and also find handles which have been deactivated
* It directly fixes javapeople.yaml and creates a PR for manual inspection

**Pull request closed without merging**